lib/src/rules/avoid_relative_lib_imports.dart

```diff
@@ -67,7 +68,8 @@ class _Visitor extends SimpleAstVisitor<void> {
     if (uriContent != null) {
       var uri = Uri.tryParse(uriContent);
       if (uri != null && uri.scheme.isEmpty) {
-        return uri.path.contains('/lib/');
+        var fullPath = node.uriSource?.fullName ?? uri.path;
+        return path.split(fullPath).contains('lib');
```

Comment:
I believe that the linter is always given a resolved AST, so it shouldn't be `null`. However, it might not be worth the risk.

@scheglov Why does the `LinterContext` expose `LinterContextUnit` instead of just exposing the `ResolvedLibraryResult`? It would solve this problem, and lint authors would learn the public API of the analyzer rather than this linter-specific API.

Reply:
I'm not very familiar with this - is adding `resourceProvider` to `LinterContext` something that needs to be done in the SDK (in package:analyzer) first?

Reply:
Yes.